kotlin/patterns-use-cases/src/main/kotlin/my/example/parallelizework/FanOutWorker.kt

```diff
@@ -0,0 +1,37 @@
+package my.example.parallelizework
+
+import dev.restate.sdk.annotation.Handler
+import dev.restate.sdk.annotation.Service
+import dev.restate.sdk.http.vertx.RestateHttpEndpointBuilder
+import dev.restate.sdk.kotlin.Awaitable
+import dev.restate.sdk.kotlin.Context
+import dev.restate.sdk.kotlin.awaitAll
+import dev.restate.sdk.kotlin.runBlock
+
+@Service
+class FanOutWorker {
+    @Handler
+    suspend fun run(ctx: Context, task: Task): TaskResult {
+        val subTasks = ctx.runBlock { task.split() }
+
+        val resultFutures: MutableList<Awaitable<SubTaskResult>> = mutableListOf()
+        for (subTask in subTasks) {
+            val subResultFuture = FanOutWorkerClient.fromContext(ctx).runSubtask(subTask)
+            resultFutures.add(subResultFuture)
+        }
+
+        val results = resultFutures.awaitAll()
```

Comment:
In kotlin you can do this way nicely as follows:

```suggestion
val results = subTasks.map {
  FanOutWorkerClient.fromContext(ctx).runSubtask(it)
}.awaitAll()
```